pkg/driver/driver_aws.go

```diff
@@ -97,25 +97,32 @@ func (d *AWSDriver) Create() (string, string, error) {
 		return "Error", "Error", err
 	}
 
-	securityGroupIDs, err := d.generateSecurityGroups(d.AWSMachineClass.Spec.NetworkInterfaces[0].SecurityGroupIDs)
-	if err != nil {
-		return "Error", "Error", err
+	var networkInterfaceSpecs []*ec2.InstanceNetworkInterfaceSpecification
+	for i, netIf := range d.AWSMachineClass.Spec.NetworkInterfaces {
+		spec := &ec2.InstanceNetworkInterfaceSpecification{
+			Groups:                   aws.StringSlice(netIf.SecurityGroupIDs),
+			DeviceIndex:              aws.Int64(int64(i)),
+			AssociatePublicIpAddress: &netIf.AssociatePublicIPAddress,
+			DeleteOnTermination:      &netIf.DeleteOnTermination,
```

Comment:
`DeleteOnTermination` will default to `false` if not specified in the MachineClass, which can potentially leave orphan resources. Can we please set `DeleteOnTermination` to `true` if not specified in the MachineClass ?
- We could also avoid setting this field at all, if not specified in the MachineClass. Please have a look at [ref](https://github.com/gardener/machine-controller-manager/blob/master/pkg/driver/driver_aws.go#L180-L195). This may also apply to `AssociatePublicIpAddress` above.

Reply:
Created the default.

Reply:
Thanks.